sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java

```diff
@@ -334,43 +348,50 @@ private void connectImpl() throws IOException
         establishWebsocketConnection();
     }
 
-    private void establishWebsocketConnection() throws IOException
-    {
+    private void establishWebsocketConnection() throws IOException {
         lastPingSent = 0;
         lastPingAck = 0;
-        WebSocketContainer client = webSocketContainerProvider.getWebSocketContainer();
+        final WebSocketContainer client = webSocketContainerProvider.getWebSocketContainer();
         final MessageHandler handler = this;
         LOGGER.debug("initiating actions to websocket");
+        if (establishConnectionThread != null) {
+            establishConnectionThread.interrupt();
+        }
+        establishConnectionThread = null;
 
-        try {
-            websocketSession = client.connectToServer(new Endpoint()
-            {
-                @Override
-                public void onOpen(Session session, EndpointConfig config)
-                {
-                    session.addMessageHandler(handler);
-                }
+        establishConnectionThread = new Thread() {
```

Comment:
Without threading it I am getting a 
`
Caused by: javax.websocket.DeploymentException
	at org.glassfish.tyrus.client.ClientManager.connectToServer(ClientManager.java:380)
	at com.ullink.slack.simpleslackapi.impl.SlackWebSocketSessionImpl.establishWebsocketConnection(SlackWebSocketSessionImpl.java:402)
	... 4 more
Caused by: java.lang.InterruptedException
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireSharedInterruptibly`

That is the reason I introduced a new thread to handle the establishWebsocketConnection behaviour